k8s/sonarqube/README.md

```diff
@@ -0,0 +1,439 @@
+```bash
+    kubectl port-forward \
+      --namespace $NAMESPACE \
+      svc/$APP_INSTANCE_NAME-sonarqube-svc \
+      9000:9000
+```
+SonarQube will be available on `localhost:9000`. All interaction with application goes  through `9000` port. Cli also will be available.
+
+To get access web-page with default credentials:
+```bash
+http://localhost:9000
+Login: admin
+Password: admin
+```
+
+# Application metrics
+
+## Prometheus metrics
+
+The application is configured to expose its metrics through [SonarQube Prometheus Exporter plugin](https://github.com/dmeiners88/sonarqube-prometheus-exporter)
+and [PostgreSQL Prometheus.io exporter plugin](https://github.com/wrouesnel/postgres_exporter)
+in the [Prometheus format](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md).
+
+Metrics can be read on a HTTP endpoint available at `[POSTGRESQL_CLUSTER_IP]:9187/metrics`,
+ and `[SONARQUBE_CLUSTER_IP]:9000/api/prometheus/metrics` where `[POSTGRESQL_CLUSTER_IP]` is the IP address of the PostgreSQL service on Kubernetes cluster, and `[SONARQUBE_CLUSTER_IP]` is IP address of service on Kubernetes cluster.
+
+## Configuring Prometheus to collect the metrics
+
+Prometheus can be configured to automatically collect the application's metrics.
+Follow the [Configuring Prometheus documentation](https://prometheus.io/docs/introduction/first_steps/#configuring-prometheus)
+to enable metrics scrapping in your Prometheus server. The detailed specification
+of `<scrape_config>` used to enable the metrics collection can be found
+[here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+
+## Exporting metrics to Stackdriver
+
+If the option to export application metrics to Stackdriver is enabled,
+the deployment includes a [`prometheus-to-sd`](https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd)
+(Prometheus to Stackdriver exporter) container.
+Then the metrics will be automatically exported to Stackdriver and visible in
+[Stackdriver Metrics Explorer](https://cloud.google.com/monitoring/charts/metrics-explorer).
+
+Each metric of the application will have a name starting with the application's name
+(matching the variable `APP_INSTANCE_NAME` described above).
+
+The exporting option might not be available for GKE on-prem clusters.
+
+> Note: Please be aware that Stackdriver has [quotas](https://cloud.google.com/monitoring/quotas)
+for the number of custom metrics created in a single GCP project. If the quota is met,
+additional metrics will not be accepted by Stackdriver, which might cause that some metrics
+from your application might not show up in the Stackdriver's Metrics Explorer.
+
+Existing metric descriptors can be removed through
+[Stackdriver's REST API](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors/delete).
+
+# Scaling
+
+SonarQube Community Edition doest not support scaling.
+
+
+# Backup and restore
```

Comment:
```suggestion
# Backup and restore

```

https://github.com/GoogleCloudPlatform/click-to-deploy/blob/f307c0ae16f8a6aa959d9c70f9a92c71b565553b/k8s/postgresql/README.md#application-metrics